crates/ruff_python_formatter/src/lib.rs

```diff
@@ -82,6 +84,18 @@ where
     }
 }
 
+/// Dummy formatter we use for rules that have not been manually implemented yet
+pub(crate) fn range_verbatim(range: TextRange, f: &mut PyFormatter) -> FormatResult<()> {
+    f.write_element(FormatElement::Tag(Tag::StartVerbatim(
+        VerbatimKind::Verbatim {
+            length: range.len(),
+        },
+    )))?;
+    write!(f, [source_text_slice(range, ContainsNewlines::Detect)])?;
+    f.write_element(FormatElement::Tag(Tag::EndVerbatim))?;
+    Ok(())
+}
```

Comment:
The idea is that this returns a struct that implements `Format` instead. It makes it nicer integrate into the formatter API

```rust
pub(crate) struct VerbatimText(TextRange);

pub(crate) const fn verbatim_text(range: TextRange) -> VerbatimText {
	VerbatimText(range)
}

impl<C> Format<C> for VerbatimText {
	fn fmt(self, &mut Formatter<C>) -> FormatResult<()> {
		// ... your code
	}
}
```

it can then be used like `write!(f, [verbatim_text(node.range())])` or `verbatim_text(node.range()).fmt(f)` if you want to avoid the macro